### DISCUSSION TITLE
tuning ipsec.conf

### DISCUSSION THREAD

**Discussion by Andrysky** [strongswan/strongswan#236] [Q&A]
hi i have a couple of questions
help me please

1. at https://wiki.strongswan.org/projects/strongswan/wiki/ConnSection I cannot find the leftauth / rightauth sections
but find in https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2StrokeExamples -> (IKEv2 Legacy Configuration Examples -> EAP_TLS authentication) -> https://www.strongswan.org/testing/testresults/ikev2/rw-eap-tls-only/moon.ipsec.conf 
1.1. leftauth / rightauth is deprecated (Legacy)?
1.2. my ubuntu "man ipsec.conf" contains sections leftauth / rightauth - wtf?

2. what's the difference between :
2.1. "EAP_TLS authentication" and "RSA authentication with X.509 certificates" [link](https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2StrokeExamples )
2.2. in  leftauth / rightauth  -> pubkey/eap-tls  - and what should i use?
2.3. IKEv2 Certificate / IKEv2 eap-tls (Certificate) - same question😅
![IMAGE 2021-03-07 18:30:10](https://user-images.githubusercontent.com/5894390/110245169-2691fb80-7f73-11eb-8cb7-146041cda82d.jpg)

**Comment by Thermi** [comment]
> hi i have a couple of questions
> help me please
> 
> 1. at https://wiki.strongswan.org/projects/strongswan/wiki/ConnSection I cannot find the leftauth / rightauth sections
>    but find in https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2StrokeExamples -> (IKEv2 Legacy Configuration Examples -> EAP_TLS authentication) -> https://www.strongswan.org/testing/testresults/ikev2/rw-eap-tls-only/moon.ipsec.conf
>    1.1. leftauth / rightauth is deprecated (Legacy)?
>    1.2. my ubuntu "man ipsec.conf" contains sections leftauth / rightauth - wtf?

It's part of the "left|rightauth" word on the wiki page.

> 2. what's the difference between :
>    2.1. "EAP_TLS authentication" and "RSA authentication with X.509 certificates" [link](https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2StrokeExamples)
>    2.2. in  leftauth / rightauth  -> pubkey/eap-tls  - and what should i use?
>    2.3. IKEv2 Certificate / IKEv2 eap-tls (Certificate) - same question

EAP_TLS uses the eap-tls method after authenticating the "server" using the pubkey method (not sure if the latter part is strictly required.
eap-tls enables mutual authentication by itself already).

IKEv2 Certificate uses the pubkey method on the client side to authenticate the server, too.

What you need to use depends on what the server is configured for.

**Reply by Andrysky** [reply]
is  leftauth(rightauth)=pubkey == leftauth(rightauth)=eap-tls ?
can you please explain why https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2StrokeExamples is a deprecated configuration? 

Following the example on the Internet, I made a root certificate and a child certificate for the server.
and config

```
config setup
         charondebug = "ike 2, knl 2, cfg 2, net 2, esp 2, dmn 2, mgr 2"
uniqueids = no

conn ikev2-vpn
     auto = add
     compress = no
     type = tunnel
     keyexchange = ikev2
     fragmentation = yes
     forceencaps = yes
     dpdaction = clear
     dpddelay = 300s
     rekey = no
     left =% any
     leftid = <id>
     leftcert = server-cert.pem
     leftsendcert = always
     leftsubnet = 0.0.0.0 / 0
     right =% any
     rightid =% any
     rightauth = eap-mschapv2
     rightsourceip = 10.10.10.0 / 24
     rightsendcert = never
     eap_identity =% identity
```

I connect successfully from my phone by login and password
now I want the phone to authenticate with a certificate too
do i need to use rightauth = pubkey or rightauth = eap-tls?

**Reply by Thermi** [reply]
> can you please explain why https://wiki.strongswan.org/projects/strongswan/wiki/IKEv2StrokeExamples is a deprecated configuration?

That's because stroke is deprecated configuration backend.

> I connect successfully from my phone by login and password
> now I want the phone to authenticate with a certificate too
> do i need to use rightauth = pubkey or rightauth = eap-tls?

Depends on what you specifically want in the end (implementation details). I use eap-tls.
